commit 24e5e65be1c9b8b433edc1723b4e13feba0db8ba
Author: Andrew Price <anprice@redhat.com>
Date:   Tue Apr 5 14:38:03 2016 +0100

    mkfs.gfs2: Open the target device with O_EXCL
    
    O_EXCL will let local mounters know that the device is busy while
    mkfs.gfs2 is running so that they don't try to access it.
    
    Before:
      # mount /dev/vdc /mnt/test
      mount: /dev/vdc: can't read superblock
    
    With O_EXCL:
      # mount /dev/vdc /mnt/test
      mount: /dev/vdc is already mounted or /mnt/test busy
    
    Resolves: rhbz#1291944
    
    Signed-off-by: Andrew Price <anprice@redhat.com>

diff --git a/gfs2/mkfs/main_mkfs.c b/gfs2/mkfs/main_mkfs.c
index b132af8..03f078e 100644
--- a/gfs2/mkfs/main_mkfs.c
+++ b/gfs2/mkfs/main_mkfs.c
@@ -558,7 +558,7 @@ void main_mkfs(int argc, char *argv[])
 
 	verify_arguments(sdp);
 
-	sdp->device_fd = open(sdp->device_name, O_RDWR | O_CLOEXEC);
+	sdp->device_fd = open(sdp->device_name, O_RDWR | O_CLOEXEC | O_EXCL);
 	if (sdp->device_fd < 0)
 		die( _("can't open device %s: %s\n"),
 		    sdp->device_name, strerror(errno));
